changeset:   1792:6949f7a75122
user:        tytso@mit.edu
date:        Tue Sep  6 06:26:45 2005 -0400
summary:     Fix blkid library so that the cache validation notices ext2 to ext3 conversion

diff -r 85a387fcad67 -r 6949f7a75122 lib/blkid/probe.c
--- a/lib/blkid/probe.c	Tue Sep  6 09:59:52 2005
+++ b/lib/blkid/probe.c	Tue Sep  6 10:26:45 2005
@@ -135,6 +135,11 @@
 	/* Distinguish between jbd and ext2/3 fs */
 	if (blkid_le32(es->s_feature_incompat) & 
 	    EXT3_FEATURE_INCOMPAT_JOURNAL_DEV)
+		return -BLKID_ERR_PARAM;
+	
+	/* Distinguish between ext3 and ext2 */
+	if ((blkid_le32(es->s_feature_compat) &
+	      EXT3_FEATURE_COMPAT_HAS_JOURNAL))
 		return -BLKID_ERR_PARAM;
 
 	get_ext2_info(dev, buf);
